lib/src/util/side_effect_visitor.dart

```diff
@@ -0,0 +1,546 @@
+  }
+
+  @override
+  visitCatchClause(CatchClause node) {
+    super.visitCatchClause(node);
+  }
+
+  @override
+  visitClassDeclaration(ClassDeclaration node) {
+    super.visitClassDeclaration(node);
+  }
+
+  @override
+  visitClassTypeAlias(ClassTypeAlias node) {
+    super.visitClassTypeAlias(node);
+  }
+
+  @override
+  visitComment(Comment node) {
+    super.visitComment(node);
+  }
+
+  @override
+  visitCommentReference(CommentReference node) {
+    super.visitCommentReference(node);
+  }
+
+  @override
+  visitCompilationUnit(CompilationUnit node) {
+    super.visitCompilationUnit(node);
+  }
+
+  @override
+  visitConditionalExpression(ConditionalExpression node) {
+    super.visitConditionalExpression(node);
+  }
+
+  @override
+  visitConfiguration(Configuration node) {
+    super.visitConfiguration(node);
+  }
+
+  @override
+  visitConstructorDeclaration(ConstructorDeclaration node) {
+    super.visitConstructorDeclaration(node);
+  }
+
+  @override
+  visitConstructorFieldInitializer(ConstructorFieldInitializer node) {
+    super.visitConstructorFieldInitializer(node);
+  }
+
+  @override
+  visitConstructorName(ConstructorName node) {
+    super.visitConstructorName(node);
+  }
+
+  @override
+  visitContinueStatement(ContinueStatement node) {
+    super.visitContinueStatement(node);
+  }
+
+  @override
+  visitDeclaredIdentifier(DeclaredIdentifier node) {
+    super.visitDeclaredIdentifier(node);
+  }
+
+  @override
+  visitDefaultFormalParameter(DefaultFormalParameter node) {
+    super.visitDefaultFormalParameter(node);
+  }
+
+  @override
+  visitDoStatement(DoStatement node) {
+    super.visitDoStatement(node);
+  }
+
+  @override
+  visitDottedName(DottedName node) {
+    super.visitDottedName(node);
+  }
+
+  @override
+  visitDoubleLiteral(DoubleLiteral node) {
+    super.visitDoubleLiteral(node);
+  }
+
+  @override
+  visitEmptyFunctionBody(EmptyFunctionBody node) {
+    super.visitEmptyFunctionBody(node);
+  }
+
+  @override
+  visitEmptyStatement(EmptyStatement node) {
+    super.visitEmptyStatement(node);
+  }
+
+  @override
+  visitEnumConstantDeclaration(EnumConstantDeclaration node) {
+    super.visitEnumConstantDeclaration(node);
+  }
+
+  @override
+  visitEnumDeclaration(EnumDeclaration node) {
+    super.visitEnumDeclaration(node);
+  }
+
+  @override
+  visitExportDirective(ExportDirective node) {
+    super.visitExportDirective(node);
+  }
+
+  @override
+  visitExpressionFunctionBody(ExpressionFunctionBody node) {
+    super.visitExpressionFunctionBody(node);
+  }
+
+  @override
+  visitExpressionStatement(ExpressionStatement node) {
+    super.visitExpressionStatement(node);
+  }
+
+  @override
+  visitExtendsClause(ExtendsClause node) {
+    super.visitExtendsClause(node);
+  }
+
+  @override
+  visitFieldDeclaration(FieldDeclaration node) {
+    super.visitFieldDeclaration(node);
+  }
+
+  @override
+  visitFieldFormalParameter(FieldFormalParameter node) {
+    super.visitFieldFormalParameter(node);
```

Comment:
I don't think you should ever visit one of these, but a field formal parameter *does* have the side effect of assigning a value to a field.